tgui/packages/tgui/interfaces/StripMenu.tsx

```diff
@@ -0,0 +1,394 @@
+    hideEmpty: true,
+  },
+
+  w_uniform: {
+    displayName: 'uniform',
+    gridSpot: getGridSpotKey([2, 1]),
+    image: 'inventory-uniform.png',
+  },
+
+  wear_suit: {
+    displayName: 'suit',
+    gridSpot: getGridSpotKey([2, 2]),
+    image: 'inventory-suit.png',
+  },
+
+  gloves: {
+    displayName: 'gloves',
+    gridSpot: getGridSpotKey([2, 3]),
+    image: 'inventory-gloves.png',
+  },
+
+  r_hand: {
+    displayName: 'right hand',
+    gridSpot: getGridSpotKey([2, 4]),
+    image: 'inventory-hand_r.png',
+    additionalComponent: <CornerText align="left">R</CornerText>,
+  },
+
+  l_hand: {
+    displayName: 'left hand',
+    gridSpot: getGridSpotKey([2, 5]),
+    image: 'inventory-hand_l.png',
+    additionalComponent: <CornerText align="right">L</CornerText>,
+  },
+
+  shoes: {
+    displayName: 'shoes',
+    gridSpot: getGridSpotKey([3, 2]),
+    image: 'inventory-shoes.png',
+  },
+
+  j_store: {
+    displayName: 'suit storage item',
+    gridSpot: getGridSpotKey([4, 0]),
+    image: 'inventory-suit_storage.png',
+  },
+
+  id: {
+    displayName: 'ID',
+    gridSpot: getGridSpotKey([4, 1]),
+    image: 'inventory-id.png',
+  },
+
+  belt: {
+    displayName: 'belt',
+    gridSpot: getGridSpotKey([4, 2]),
+    image: 'inventory-belt.png',
+  },
+
+  back: {
+    displayName: 'backpack',
+    gridSpot: getGridSpotKey([4, 3]),
+    image: 'inventory-back.png',
+  },
+
+  l_store: {
+    displayName: 'left pocket',
+    gridSpot: getGridSpotKey([4, 4]),
+    image: 'inventory-pocket.png',
+  },
+
+  r_store: {
+    displayName: 'right pocket',
+    gridSpot: getGridSpotKey([4, 5]),
+    image: 'inventory-pocket.png',
+  },
+};
+
+enum ObscuringLevel {
+  Completely = 1,
+  Hidden = 2,
+}
+
+type Interactable = {
+  interacting: BooleanLike;
+};
+
+/**
+ * Some possible options:
+ *
+ * null - No interactions, no item, but is an available slot
+ * { interacting: 1 } - No item, but we're interacting with it
+ * { icon: icon, name: name } - An item with no alternate actions
+ *   that we're not interacting with.
+ * { icon, name, interacting: 1 } - An item with no alternate actions
+ *   that we're interacting with.
+ */
+type StripMenuItem =
+  | null
+  | Interactable
+  | ((
+      | {
+          icon: string;
+          name: string;
+          alternate: string;
+        }
+      | {
+          obscured: ObscuringLevel;
+        }
+    ) &
+      Partial<Interactable>);
+
+type StripMenuData = {
+  items: Record<keyof typeof SLOTS, StripMenuItem>;
+  name: string;
+};
+
+export const StripMenu = (props, context) => {
+  const { act, data } = useBackend<StripMenuData>();
+
+  const gridSpots = new Map<GridSpotKey, string>();
+  for (const key of Object.keys(data.items)) {
+    const item = data.items[key];
+    if (item === null && SLOTS[key].hideEmpty) continue;
+    gridSpots.set(SLOTS[key].gridSpot, key);
+  }
+
+  return (
+    <Window title={`Stripping ${data.name}`} width={400} height={400}>
```

Comment:
This component is pretty big and has a lot of different outcomes. Can this be broken down? At the very least at the columns map - we can have a new component just to represent a single item to be placed.

Reply:
Can you elaborate a bit? My tgui knowledge is a bit lacking so I am not entirely sure what do you want me to do and how

Reply:
Yeah sure. The issue is that we have one large component that does everything. From a code structure and performance POV we can break this down into smaller components. From a code POV we can reduce the levels of indentation and some of the complexity.

For example - when the columns are being mapped over we can have a component to render that element, rather than having everything together. If this is clear as mud then let me know as I don't think it's the best explanation

Reply:
Yeah, sorry, cannot really wrap my head around it. I think I understand the issue, but I don't understand how to solve it, how to split the menu component into columns and make them somehow render separately. Sorry.

Reply:
It's all good. We can take this as is and always upgrade the standard of the code. This is fundamentally a good improvement.